app/controllers/users_controller.rb

```diff
@@ -1,30 +1,77 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user, only: %i[:edit, :update]
+  def index
+    @users = User.paginate page: params[:page]
+  end
+
   def new
     @user = User.new
   end
 
   def show
-    @user = User.find_by id: params[:id]
-    return if @user.present?
-
-    flash[:danger] = t"new.not_found", id: params[:id]
-    redirect_to :home
+    check_user
   end
 
   def create
     @user = User.new user_params
     if @user.save
-      flash[:success] = t"new.title"
+      flash[:success] = t "new.title"
       log_in @user
       redirect_to @user
     else
       render :new
     end
   end
 
+  def edit
+    check_user
+  end
+
+  def update
+    check_user
+    if @user.update(user_params)
+      flash[:success] = t "user.update_success"
+      redirect_to @user
+    else
+      render "edit"
+    end
+  end
+
+  def destroy
+    check_user
+    if @user&.destroy
```

Comment:
ở trước đó đã có xử lý check_user rồi thì chỗ này ko cần check lại nữa đâu e nhé
```suggestion
    if @user.destroy
```